src/workspace/mod.rs

```diff
@@ -149,6 +163,15 @@ impl Workspace {
             new_path.push(package.manifest_path.strip_prefix(&self.workspace_root)?);
             let new_manifest = ManifestPath::new(new_path)?;
 
+            // replace the original path to the package for which metadata should be
+            // generated with the absolute path to the package in the temporary directory.
+            if let Some(package) = manifest.metadata_package_mut() {
+                let absolute_package_path = package.path.canonicalize()?;
+                let mut new_path: PathBuf = target.as_ref().into();
+                new_path.push(absolute_package_path.strip_prefix(&self.workspace_root)?);
+                package.path = new_path;
+            }
```

Comment:
I've moved this logic here now, since the block above does basically the same thing for the manifest path.

Also we would need to additionally pass `self.workspace_root` and the `target` directory into `manifest` if we would really like to put the logic there.

Reply:
It does remove the hardcoded relative path, but this pattern doesn't feel 100% right because the construction of this path is split across `cmd/metadata.rs` and `workspace/mod.rs`, then finally used in `manifest.rs`. If encapsulating it means passing extra context into `manifest` via fields or method arguments, then it should be considered (if practical).